File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/git_rebase_details.py
# coding: utf-8
# Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.



from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class GitRebaseDetails(object):
    """
    Details needed to perform a git rebase for a folder context.
    """

    def __init__(self, **kwargs):
        """
        Initializes a new GitRebaseDetails object with values from keyword arguments.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param git_folder_path:
            The value to assign to the git_folder_path property of this GitRebaseDetails.
        :type git_folder_path: str

        :param branch_name:
            The value to assign to the branch_name property of this GitRebaseDetails.
        :type branch_name: str

        :param remote_branch_name:
            The value to assign to the remote_branch_name property of this GitRebaseDetails.
        :type remote_branch_name: str

        :param commit_id:
            The value to assign to the commit_id property of this GitRebaseDetails.
        :type commit_id: str

        """
        self.swagger_types = {
            'git_folder_path': 'str',
            'branch_name': 'str',
            'remote_branch_name': 'str',
            'commit_id': 'str'
        }

        self.attribute_map = {
            'git_folder_path': 'gitFolderPath',
            'branch_name': 'branchName',
            'remote_branch_name': 'remoteBranchName',
            'commit_id': 'commitId'
        }

        self._git_folder_path = None
        self._branch_name = None
        self._remote_branch_name = None
        self._commit_id = None

    @property
    def git_folder_path(self):
        """
        Gets the git_folder_path of this GitRebaseDetails.
        Folder path used to locate the git worktree.


        :return: The git_folder_path of this GitRebaseDetails.
        :rtype: str
        """
        return self._git_folder_path

    @git_folder_path.setter
    def git_folder_path(self, git_folder_path):
        """
        Sets the git_folder_path of this GitRebaseDetails.
        Folder path used to locate the git worktree.


        :param git_folder_path: The git_folder_path of this GitRebaseDetails.
        :type: str
        """
        self._git_folder_path = git_folder_path

    @property
    def branch_name(self):
        """
        Gets the branch_name of this GitRebaseDetails.
        The name of the git branch


        :return: The branch_name of this GitRebaseDetails.
        :rtype: str
        """
        return self._branch_name

    @branch_name.setter
    def branch_name(self, branch_name):
        """
        Sets the branch_name of this GitRebaseDetails.
        The name of the git branch


        :param branch_name: The branch_name of this GitRebaseDetails.
        :type: str
        """
        self._branch_name = branch_name

    @property
    def remote_branch_name(self):
        """
        **[Required]** Gets the remote_branch_name of this GitRebaseDetails.
        Remote branch name to rebase onto (optional).


        :return: The remote_branch_name of this GitRebaseDetails.
        :rtype: str
        """
        return self._remote_branch_name

    @remote_branch_name.setter
    def remote_branch_name(self, remote_branch_name):
        """
        Sets the remote_branch_name of this GitRebaseDetails.
        Remote branch name to rebase onto (optional).


        :param remote_branch_name: The remote_branch_name of this GitRebaseDetails.
        :type: str
        """
        self._remote_branch_name = remote_branch_name

    @property
    def commit_id(self):
        """
        Gets the commit_id of this GitRebaseDetails.
        Commit id to rebase onto (optional).


        :return: The commit_id of this GitRebaseDetails.
        :rtype: str
        """
        return self._commit_id

    @commit_id.setter
    def commit_id(self, commit_id):
        """
        Sets the commit_id of this GitRebaseDetails.
        Commit id to rebase onto (optional).


        :param commit_id: The commit_id of this GitRebaseDetails.
        :type: str
        """
        self._commit_id = commit_id

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other

File: aidp-java-client-2x/src/main/java/com/oracle/aidataplatform/dp/model/GitRebaseDetails.java
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.

package com.oracle.aidataplatform.dp.model;


/**
 * Details needed to perform a Git rebase for a folder context.
**/
@javax.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=GitRebaseDetails.Builder.class)

public final class GitRebaseDetails  {
    @Deprecated
    @java.beans.ConstructorProperties({"gitFolderPath", "branchName", "remoteBranchName", "commitId"})
    public GitRebaseDetails(String gitFolderPath, String branchName, String remoteBranchName, String commitId) {
        super();
        this.gitFolderPath = gitFolderPath;
        this.branchName = branchName;
        this.remoteBranchName = remoteBranchName;
        this.commitId = commitId;
    }

    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
            /**
     * Folder path used to locate the Git worktree.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("gitFolderPath")
        private String gitFolderPath;

                /**
         * Folder path used to locate the Git worktree.
         * @param gitFolderPath the value to set
         * @return this builder
         **/
        

        public Builder gitFolderPath(String gitFolderPath) {
        this.gitFolderPath = gitFolderPath;
        return this;
        }
            /**
     * The name of the Git branch.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("branchName")
        private String branchName;

                /**
         * The name of the Git branch.
         * @param branchName the value to set
         * @return this builder
         **/
        

        public Builder branchName(String branchName) {
        this.branchName = branchName;
        return this;
        }
            /**
     * Remote branch name to rebase onto (optional).
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("remoteBranchName")
        private String remoteBranchName;

                /**
         * Remote branch name to rebase onto (optional).
         * @param remoteBranchName the value to set
         * @return this builder
         **/
        

        public Builder remoteBranchName(String remoteBranchName) {
        this.remoteBranchName = remoteBranchName;
        return this;
        }
            /**
     * Commit ID to rebase onto (optional).
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("commitId")
        private String commitId;

                /**
         * Commit ID to rebase onto (optional).
         * @param commitId the value to set
         * @return this builder
         **/
        

        public Builder commitId(String commitId) {
        this.commitId = commitId;
        return this;
        }


        public GitRebaseDetails build() {
            GitRebaseDetails model = new GitRebaseDetails(this.gitFolderPath
                    , this.branchName
                    , this.remoteBranchName
                    , this.commitId);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(GitRebaseDetails model) {
                this.gitFolderPath(model.getGitFolderPath());
    this.branchName(model.getBranchName());
    this.remoteBranchName(model.getRemoteBranchName());
    this.commitId(model.getCommitId());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    


        /**
     * Folder path used to locate the Git worktree.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("gitFolderPath")
    private final String gitFolderPath;

        /**
     * Folder path used to locate the Git worktree.
     * @return the value
     **/
    
    public String getGitFolderPath() {
        return gitFolderPath;
    }


        /**
     * The name of the Git branch.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("branchName")
    private final String branchName;

        /**
     * The name of the Git branch.
     * @return the value
     **/
    
    public String getBranchName() {
        return branchName;
    }


        /**
     * Remote branch name to rebase onto (optional).
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("remoteBranchName")
    private final String remoteBranchName;

        /**
     * Remote branch name to rebase onto (optional).
     * @return the value
     **/
    
    public String getRemoteBranchName() {
        return remoteBranchName;
    }


        /**
     * Commit ID to rebase onto (optional).
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("commitId")
    private final String commitId;

        /**
     * Commit ID to rebase onto (optional).
     * @return the value
     **/
    
    public String getCommitId() {
        return commitId;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("GitRebaseDetails(");
        sb.append("gitFolderPath=").append(String.valueOf(this.gitFolderPath));
        sb.append(", branchName=").append(String.valueOf(this.branchName));
        sb.append(", remoteBranchName=").append(String.valueOf(this.remoteBranchName));
        sb.append(", commitId=").append(String.valueOf(this.commitId));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof GitRebaseDetails)) {
            return false;
        }

        GitRebaseDetails other = (GitRebaseDetails) o;
        return java.util.Objects.equals(this.gitFolderPath, other.gitFolderPath) &&
            java.util.Objects.equals(this.branchName, other.branchName) &&
            java.util.Objects.equals(this.remoteBranchName, other.remoteBranchName) &&
            java.util.Objects.equals(this.commitId, other.commitId);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = 1;
        result = (result * PRIME) + (this.gitFolderPath == null ? 43 : this.gitFolderPath.hashCode());
        result = (result * PRIME) + (this.branchName == null ? 43 : this.branchName.hashCode());
        result = (result * PRIME) + (this.remoteBranchName == null ? 43 : this.remoteBranchName.hashCode());
        result = (result * PRIME) + (this.commitId == null ? 43 : this.commitId.hashCode());
        return result;
    }


}
